File: KoKoslan/src/kokoslan/java/ast/KoKoProgram.java
/**
 @author loriacarlos@gmail.com
 
*/
package kokoslan.java.ast;
import java.util.*;
import java.io.*;
import kokoslan.java.eval.*;

public class KoKoProgram implements KoKoAst{
   private List<KoKoAst> statements;
   
   public KoKoProgram(List<KoKoAst> statements){
      this.statements = statements;
	  
   }
   public void genCode(PrintStream out){
       this.statements.stream()
	                  .forEach( t -> t.genCode(out));
   }
   public KoKoValue eval(KoKoContext ctx){
		KoKoValue res = null;
		for(int i = 0; i < statements.size() -1 ; i++){
			res = statements.get(i).eval(ctx);
		}
		res = statements.get(statements.size() -1).eval(ctx);
		return res;
		
	}
	public KoKoValue eval(){
		return  eval(new KoKoContext());
	}
}

File: KoKoslan/src/kokoslan/kotlin/ast/KoKoProgram.kt
/** 
   Andrey Arguedas Espinoza
   Daniela Armas Sanchez
   Michael Chen Wang
   Kimberly Olivas Delgado
*/

package kokoslan.kotlin.ast

import java.io.*
import kokoslan.kotlin.eval.*

class KoKoProgram(private val statements: List < KoKoAst > ): KoKoAst {

    override fun genCode(Out: PrintStream) {
        statements.forEach {
            it.genCode(Out)
        }
    }

    override fun eval(ctx: KoKoContext): KoKoValue? {
        var res: KoKoValue?
        (0 until statements.size - 1)
                .map { statements.get(it) }
                .forEach { res = it.eval(ctx) }
        res = statements[statements.size - 1].eval(ctx)
        return res
    }

    fun eval(): KoKoValue? {
        return eval(KoKoContext())
    }
    
}